src/interface.ts

```diff
@@ -67,7 +67,7 @@ export interface RenderedCell<RecordType> {
 
 export type Direction = 'ltr' | 'rtl';
 
-export type DataIndex<T = any> = DeepNamePath<T> | '';
+export type DataIndex<T = any> = DeepNamePath<T> | (string | {});
```

Comment:
```suggestion
type SpecialString<T extends string> = T | (string & {});
export type DataIndex<T = any> = DeepNamePath<T> | (string | {});
```

type DataIndex<T = any> = DeepNamePath<T> | SpecialString<DeepNamePath<T>>